### PR TITLE
Add Nascent keyboard support.

### PR DESCRIPTION
**Pull request by studiokestra**
## Description

QMK firmware support for the Nascent keyboard by Studio Kestra.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/studiokestra/nascent/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/studiokestra/nascent/rules.mk:18]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/studiokestra/nascent/rules.mk (hunk 1 above)
Comment (studiokestra) [reply keyboards/studiokestra/nascent/rules.mk]: Makes sense, done.

**Review by drashna — APPROVED** [review]

File: keyboards/studiokestra/nascent/info.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
+    "keyboard_name": "Nascent",
+    "url": "https://studiokestra.ca/",
+    "maintainer": "Studio Kestra",
+    "width": 16,
+    "height": 5,
```
Comment (fauxpark) [inline keyboards/studiokestra/nascent/info.json:6]:
```suggestion
```

File: keyboards/studiokestra/nascent/rules.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/studiokestra/nascent/rules.mk:18]:
```suggestion
```

File: keyboards/studiokestra/nascent/rules.mk [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/studiokestra/nascent/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by studiokestra** [comment]
Additional requested changes have been made, thanks.

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]
As this keyboard has missed the first pass of Data Driven info.json PRs (#17865) would you kindly consider moving as much configuration as you can to info.json- reference here [data driven config](https://docs.qmk.fm/#/data_driven_config)

File: keyboards/studiokestra/nascent/info.json [hunk 5]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "Nascent",
+    "url": "https://studiokestra.ca/",
+    "maintainer": "Studio Kestra",
```
Comment (fauxpark) [inline keyboards/studiokestra/nascent/info.json:4]:
```suggestion
    "keyboard_name": "Nascent",
    "manufacturer": "Studio Kestra",
    "url": "https://studiokestra.ca/",
    "maintainer": "Studio Kestra",
    "usb": {
        "vid": "0x7C10",
        "pid": "0x0165",
        "device_version": "0.0.1"
    },
```
I haven't actually done the S boards yet, so this is fine, but might as well do it here.

File: keyboards/studiokestra/nascent/config.h [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x7C10
+#define PRODUCT_ID      0x0165
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Studio Kestra
+#define PRODUCT         Nascent
+
```
Comment (fauxpark) [inline keyboards/studiokestra/nascent/config.h:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/studiokestra/nascent/info.json (hunk 5 above)
Comment (keyboard-magpie) [reply keyboards/studiokestra/nascent/info.json:4]: Sort order in my head was different, sorry!

**Comment by studiokestra** [comment]
Have made suggested changes. thanks

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**